core/src/main/java/org/apache/commons/vfs2/provider/AbstractFileSystem.java

```diff
@@ -317,42 +317,46 @@ public FileObject resolveFile(final FileName name) throws FileSystemException
         return resolveFile(name, true);
     }
 
-    private synchronized FileObject resolveFile(final FileName name, final boolean useCache) throws FileSystemException
```

Comment:
thats a quite delicate location/change. Can you somehow reproduce the deadlocks or show thread dumps?